mosqagent.c

```diff
@@ -105,6 +107,100 @@ void mqtta_dispose_message(struct mqtta_message *msg)
     free(msg);
 }
 
+/**
+ * If there is already a configuration in the agent, make sure to take care
+ * of the memory object before calling this function, otherwise there may be
+ * memory leaks!
+ */
+int mqtta_load_configuration(struct mosqagent *agent,
+                             const char* filepath)
+{
+    int ret = 0;
+
+    struct mosqagent_config *config;
+
+    config_t configuration;
+
+    const char* client_name;
+    const char* broker_host;
+
+    // Init the configuration struct from libconfig
+    config_init(&configuration);
+
+    /*
+     * Check this part first. Not finding the config file is the most likely
+     * failure and we'll fail this before reserving memory.
+     */
+
+    if (!config_read_file(&configuration, filepath)) {
+        fprintf(stderr, "Cannot read config file: %s\n", config_error_text(&configuration));
+        ret = 1;
+        goto cleanup_with_configuration;
```

Comment:
Documentation of those return values is missing. define or enum in a central place would be nice for those.